test/groovy/BuildPluginStepTests.groovy

```diff
@@ -77,7 +77,7 @@ class BuildPluginStepTests extends BaseTest {
     def script = loadScript(scriptName)
     def configurations = script.getConfigurations(platforms: ['bar', 'foo'])
     println configurations
-    assertEquals(configurations.size, 2)
+    assertEquals(configurations.size(), 2)
```

Comment:
what is the reason of this change?

Reply:
the unit test on that part of the code was broken ... missing those 2 () ... I discovered this by chance and took the opportunity to correct it ... part of why this is in draft, I planned to discuss this with you tomorrow.

Reply:
As discussed together:
- The unit tests are failing when Java 17 is used. Since the requirement here is Java 11, this change is out of scope for this PR
- This change (`s/size/size()`)  is correct and easier to read for non Groovy specialist: might be worth it to add it on another PR, along with enabling the skipped test (or deleting it)